### PR TITLE
Don't bother cleaning up observers if we're shutting down. r=johannh

### PR DESCRIPTION
**Pull request by nhnt11**
This is an attempt to mitigate a shutdown crash, see https://bugzilla.mozilla.org/show_bug.cgi?id=1464938

### CODE REVIEW

File: src/privileged/api.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ this.fxmonitor = class extends ExtensionAPI {
   }
 
   onShutdown(shutdownReason) {
-    if (!FirefoxMonitorContainer.FirefoxMonitor) {
+    if (Services.startup.shuttingDown || !FirefoxMonitorContainer.FirefoxMonitor) {
```
Comment (aswan) [inline src/privileged/api.js]: sorry for the drive-by, you can also just check if `shutdownReason == "APP_SHUTDOWN"`

Comment (nhnt11) [reply src/privileged/api.js]: I purposely did not do this, because, for example, if the addon is installed as a temporary add-on, the reason is ADDON_UNINSTALLED. And we *do* want to handle that case if we're not shutting down.

Comment (nhnt11) [reply src/privileged/api.js]: It's also ADDON_UNINSTALLED (or disabled) if there is a pending disabled.

In the real world since this is being shipped as a system add-on, this shouldn't matter, but for hygiene and resilience to shipping strategy I prefer to just use Services.startup.shuttingDown here.
